jetty-server/src/test/java/org/eclipse/jetty/server/GracefulStopTest.java

```diff
@@ -55,6 +56,166 @@
 @RunWith(AdvancedRunner.class)
 public class GracefulStopTest 
 {
+    /**
+     * Test of standard graceful timeout mechanism when a block request does
+     * not complete
+     * @throws Exception on test failure
+     */
+    @Test
+    public void testSleepNotGraceful() throws Exception
```

Comment:
Can these 3 new tests be parametrized ? They share 99% of the same code, so perhaps it's possible to just call a private method with 2-3 parameters to control its behavior ?

Reply:
sure